internal/service/ssoadmin/customer_managed_policy_attachment.go

```diff
@@ -0,0 +1,247 @@
+}
+
+func resourceCustomerManagedPolicyAttachmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).SSOAdminConn
+
+	policyName, policyPath, permissionSetArn, instanceArn, err := ParseCustomerManagedPolicyAttachmentID(d.Id())
+	if err != nil {
+		return fmt.Errorf("error parsing SSO Customer Managed Policy Attachment ID: %w", err)
+	}
+
+	policy, err := FindCustomerManagedPolicy(conn, policyName, policyPath, permissionSetArn, instanceArn)
+
+	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeResourceNotFoundException) {
+		log.Printf("[WARN] Customer Managed Policy (%s) for SSO Permission Set (%s) not found, removing from state", policyName, permissionSetArn)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading Customer Managed Policy (%s) for SSO Permission Set (%s): %w", policyName, permissionSetArn, err)
+	}
+
+	if policy == nil {
+		log.Printf("[WARN] Customer Managed Policy (%s) for SSO Permission Set (%s) not found, removing from state", policyName, permissionSetArn)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("instance_arn", instanceArn)
+	d.Set("permission_set_arn", permissionSetArn)
+	d.Set("customer_managed_policy_reference", flattenPolicyReference(policy))
+
+	return nil
+}
+
+func resourceCustomerManagedPolicyAttachmentDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).SSOAdminConn
+
+	policyName, policyPath, permissionSetArn, instanceArn, err := ParseCustomerManagedPolicyAttachmentID(d.Id())
+	if err != nil {
+		return fmt.Errorf("error parsing SSO Customer Managed Policy Attachment ID: %w", err)
+	}
+
+	input := &ssoadmin.DetachCustomerManagedPolicyReferenceFromPermissionSetInput{
+		InstanceArn:      aws.String(instanceArn),
+		PermissionSetArn: aws.String(permissionSetArn),
+		CustomerManagedPolicyReference: &ssoadmin.CustomerManagedPolicyReference{
+			Name: aws.String(policyName),
+			Path: aws.String(policyPath),
+		},
+	}
+	err = resource.Retry(customerPolicyAttachmentTimeout, func() *resource.RetryError {
+		var err error
+		_, err = conn.DetachCustomerManagedPolicyReferenceFromPermissionSet(input)
+
+		if err != nil {
+			if tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeConflictException) {
+				return resource.RetryableError(err)
+			}
+			if tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeThrottlingException) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+
+	})
+
+	if tfresource.TimedOut(err) {
+		_, err = conn.DetachCustomerManagedPolicyReferenceFromPermissionSet(input)
+	}
+
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, ssoadmin.ErrCodeResourceNotFoundException) {
+			return nil
+		}
+		return fmt.Errorf("error detaching Customer Managed Policy (%s) from SSO Permission Set (%s): %w", policyName, permissionSetArn, err)
+	}
+
+	// Provision ALL accounts after detaching the managed policy
```

Comment:
Same comment as earlier, lets be a bit more specific on what/why we're doing exactly here.